breaking.md cleanup - release/v2int/1.2 (#12169)
microsoft/FluidFramework: BREAKING.md

# Adding breaking and upcoming change notes

Notes on breaking, upcoming, and otherwise interesting changes go here. They will be reviewed and published along with each release.  Published changelogs may be found on the docs site at fluidframework.com.

> For more information on best practices around communicating breaking please see our docs [here](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes)

Reviewer: nit to sound more authoritative:
```suggestion
For instructions on how to communicate breaking changes please see our docs [here](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes)
```

- Avoid using code formatting in the title (it's fine to use in the body).
- To explain the benefit of your change, use the [What's New](https://fluidframework.com/docs/updates/v1.0.0/) section on FluidFramework.com.

> # 2.0.0-internal-1.1.0

Reviewer: Isn't it `internal.1.1`, not `internal-1.1`?

Reviewer: Also... should this section (for `next` stuff) be about `2.0.0-internal-2.0.0`?  i.e. the next major bump?

Author: > Also... should this section (for `next` stuff) be about `2.0.0-internal-2.0.0`? i.e. the next major bump?

Yes, everything under the `2.0.0-internal-1.0.0 Upcoming changes` should be actionable items that you can take for to prepare for breaking changes in a future major release

[More in-depth explanation](https://github.com/microsoft/FluidFramework/wiki/Communicating-breaking-changes#breaking-changes-vs-upcoming-changes)

Reviewer: Ok I see now that minor bumps get Upcoming sections.  I would lean towards using the major version in the H1 heading.

Or maybe `# 2.0.0-internal.1.x` or `# 2.0.0-internal.1.0.0 through 2.0.0-internal.1.1.0` and update it each release.  I'd prefer the former (`1.x`)

## 2.0.0-internal-1.1.0 Upcoming changes
- [Add assertion that prevents sending op while processing another op](#add-assertion-that-prevents-sending-op-while-processing-another-op)
- [Remove type field from ShareLinkInfoType](#Remove-type-field-from-ShareLinkInfoType)
- [Remove ShareLinkTypes interface](#Remove-ShareLinkTypes-interface)
- [Remove enableShareLinkWithCreate from HostStoragePolicy](#Remove-enableShareLinkWithCreate-from-HostStoragePolicy)

> - [Various return types in @fluidframework/sequence have been widened to include undefined](#various-return-types-in-fluidframeworksequence-have-been-widened-to-include-undefined)

Reviewer: This looks new - I'd pull that into a separate PR after this cleanup

+       createShareLinkType?: ShareLinkTypes | ISharingLinkKind,
    ):
```
### Remove enableShareLinkWithCreate from HostStoragePolicy
`enableShareLinkWithCreate` feature gate has been deprecated and will be removed in a future breaking change. If you wish to enable creation of a sharing link along with the creation of Fluid file, you will need to provide `createShareLinkType:ISharingLinkKind` input to the `createOdspCreateContainerRequest` function and enable the feature using `enableSingleRequestForShareLinkWithCreate` in `HostStoragePolicy`

# 2.0.0-internal-1.0.0

> ## 2.0.0-internal-1.0.0 Upcoming changes

Reviewer: For future consideration: I'm doing some brainstorming about how the Upcoming changes are organized/presented.

1. Do we want the "upcoming" section to be tied to a version?
    * Pro of mentioning a version is it may be required to be on at least that version to get the alternative to migrate to.  If this is the common/typical case, then probably leave it.
    * Con is that this information gets spread throughout this file, which may make it harder to keep up with the list.  I'm also confused about if there'd be Upcoming sections for minor releases - looks like there are.

2. In light of this realization, maybe we want a separate file, so this info is not interleaved with true Breaking changes, especially since it's updated on a different cadence.  And I imagine there could also be weirdness keeping Upcoming change notes in sync between main and next, whereas that wouldn't so much be the case with Breaking (which would only be added to next, right?).

Reviewer: cc @scottn12

`ConnectionState.Connecting` is marked as deprecated, please use `ConnectionState.CatchingUp` instead.
`ConnectionState.Connecting` will be removed in the following major release.

> ### Remove ICodeLoader from `@fluidframework/container-definitions`

Reviewer: This is suspicious... Did we not do what this says?  If it was reverted I would leave it in with a big bold note saying it was reverted.  Maybe strikethrough the text (I did this with bindToContext last week)

Author: This note was actually moved, not deleted. It was originally placed into the wrong section and was moved into the correct one in this PR. You can find this note on line 141 now
